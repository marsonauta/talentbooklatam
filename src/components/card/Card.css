.latamer {
    float: left;
    border: 1px solid #333;
    padding:24px;
    border-radius:4px;
    background:all .5s ease;
  }
  .latamer:hover{
    background:hsla(237,90%,49%,.03);
    transition:background .5s ease;
    background:all .5s ease;
  }
  .latamer p{
    margin-bottom:16px;
  }