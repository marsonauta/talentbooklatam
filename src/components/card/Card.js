import React from 'react'
import './Card.css'

const Card = (props) => {
    return <article className={props.rolClass} >
        <h3>{props.name}</h3>
        <h4>{props.rol}</h4>
        <p>{props.summary}</p>
        <ul className="contact">
            <li className="contact-cv"><a href="https://drive.google.com/open?id=1E48rb5mKkxbJkkgL8lZWurvk1cwHn_St" target="_blank"><i className="fas fa-passport"></i> Curr√≠culum</a></li>
            <li className="contact-in"><a href="https://www.linkedin.com/in/andreaescudero/" target="_blank"><i className="fab fa-linkedin"></i> LinkedIn</a></li>
            <li className="contact-wl"><a href="https://medium.com/latam-airlines/accesibilidad-digital-creando-experiencias-para-todos-52e420d3dbdb" target="_blank"><i className="fab fa-medium"></i> Medium</a></li>
        </ul>
    </article>
}

export default Card