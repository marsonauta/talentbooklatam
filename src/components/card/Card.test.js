import React from 'react';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Card from './Card'

describe('Card component', () => {
    test('render with specific class', () => {
        const expectedClass = "latamer uxd senior"

        const { container } = render(<Card rolClass={expectedClass}/>)
        expect(container.firstChild).toMatchSnapshot(`<article class="${expectedClass}"`)
    }) 

    test('render the name', () => {
        const expectedName = "Andrea Escudero"

        const { container, getByText } = render(<Card name={expectedName}/>)
        expect(getByText(expectedName)).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot(`<h3>${expectedName}</h3>`)
    })

    test('render the role', () => {
        const expectedRol = "UX Designer Senior"

        const { container, getByText } = render(<Card rol={expectedRol} />)
        expect(getByText(expectedRol)).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot(`<h4>${expectedRol}</h4>`)
    })

    test('render the summary', () => {
        const expectedSummary = "My summary"

        const { container, getByText } = render(<Card summary={expectedSummary} />)
        expect(getByText(expectedSummary)).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot(`<p>${expectedSummary}</p>`)
    })
})


