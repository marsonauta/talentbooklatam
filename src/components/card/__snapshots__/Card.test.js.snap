// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component render the name: <h3>Andrea Escudero</h3> 1`] = `
<article>
  <h3>
    Andrea Escudero
  </h3>
  <h4 />
  <p />
  <ul
    class="contact"
  >
    <li
      class="contact-cv"
    >
      <a
        href="https://drive.google.com/open?id=1E48rb5mKkxbJkkgL8lZWurvk1cwHn_St"
        target="_blank"
      >
        <i
          class="fas fa-passport"
        />
         Currículum
      </a>
    </li>
    <li
      class="contact-in"
    >
      <a
        href="https://www.linkedin.com/in/andreaescudero/"
        target="_blank"
      >
        <i
          class="fab fa-linkedin"
        />
         LinkedIn
      </a>
    </li>
    <li
      class="contact-wl"
    >
      <a
        href="https://medium.com/latam-airlines/accesibilidad-digital-creando-experiencias-para-todos-52e420d3dbdb"
        target="_blank"
      >
        <i
          class="fab fa-medium"
        />
         Medium
      </a>
    </li>
  </ul>
</article>
`;

exports[`Card component render the role: <h4>UX Designer Senior</h4> 1`] = `
<article>
  <h3 />
  <h4>
    UX Designer Senior
  </h4>
  <p />
  <ul
    class="contact"
  >
    <li
      class="contact-cv"
    >
      <a
        href="https://drive.google.com/open?id=1E48rb5mKkxbJkkgL8lZWurvk1cwHn_St"
        target="_blank"
      >
        <i
          class="fas fa-passport"
        />
         Currículum
      </a>
    </li>
    <li
      class="contact-in"
    >
      <a
        href="https://www.linkedin.com/in/andreaescudero/"
        target="_blank"
      >
        <i
          class="fab fa-linkedin"
        />
         LinkedIn
      </a>
    </li>
    <li
      class="contact-wl"
    >
      <a
        href="https://medium.com/latam-airlines/accesibilidad-digital-creando-experiencias-para-todos-52e420d3dbdb"
        target="_blank"
      >
        <i
          class="fab fa-medium"
        />
         Medium
      </a>
    </li>
  </ul>
</article>
`;

exports[`Card component render the summary: <p>My summary</p> 1`] = `
<article>
  <h3 />
  <h4 />
  <p>
    My summary
  </p>
  <ul
    class="contact"
  >
    <li
      class="contact-cv"
    >
      <a
        href="https://drive.google.com/open?id=1E48rb5mKkxbJkkgL8lZWurvk1cwHn_St"
        target="_blank"
      >
        <i
          class="fas fa-passport"
        />
         Currículum
      </a>
    </li>
    <li
      class="contact-in"
    >
      <a
        href="https://www.linkedin.com/in/andreaescudero/"
        target="_blank"
      >
        <i
          class="fab fa-linkedin"
        />
         LinkedIn
      </a>
    </li>
    <li
      class="contact-wl"
    >
      <a
        href="https://medium.com/latam-airlines/accesibilidad-digital-creando-experiencias-para-todos-52e420d3dbdb"
        target="_blank"
      >
        <i
          class="fab fa-medium"
        />
         Medium
      </a>
    </li>
  </ul>
</article>
`;

exports[`Card component render with specific class: <article class="latamer uxd senior" 1`] = `
<article
  class="latamer uxd senior"
>
  <h3 />
  <h4 />
  <p />
  <ul
    class="contact"
  >
    <li
      class="contact-cv"
    >
      <a
        href="https://drive.google.com/open?id=1E48rb5mKkxbJkkgL8lZWurvk1cwHn_St"
        target="_blank"
      >
        <i
          class="fas fa-passport"
        />
         Currículum
      </a>
    </li>
    <li
      class="contact-in"
    >
      <a
        href="https://www.linkedin.com/in/andreaescudero/"
        target="_blank"
      >
        <i
          class="fab fa-linkedin"
        />
         LinkedIn
      </a>
    </li>
    <li
      class="contact-wl"
    >
      <a
        href="https://medium.com/latam-airlines/accesibilidad-digital-creando-experiencias-para-todos-52e420d3dbdb"
        target="_blank"
      >
        <i
          class="fab fa-medium"
        />
         Medium
      </a>
    </li>
  </ul>
</article>
`;
